libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/LUResource.cs

```diff
@@ -0,0 +1,407 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma warning disable CA2227
+#pragma warning disable CA1716
+#pragma warning disable SA1402
+#pragma warning disable SA1601
+#pragma warning disable SA1602
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Microsoft.Bot.Builder.Parsers.LU.Parser
+{
+    public enum SectionType
+    {
+        SimpleIntentSection,
+        NestedIntentSection,
+        EntitySection,
+        NewEntitySection,
+        ImportSection,
+        ModelInfoSection,
+        QnaSection
+    }
+
+    public enum TypeEnum
+    {
+        Intent,
+        Entities,
+        PatternAnyEntities,
+        ClosedLists,
+        Prebuilt,
+        Utterance,
+        Patterns,
+        Regex,
+        Composites,
+        MachineLearned
+    }
+
+    public partial class LuResource
+    {
+        public LuResource(List<Section> sections, string content, List<Error> errors)
```

Comment:
Is the .Net convention still to take "IList<Error>" as the type here, rather than List<T>?

Reply:
Further more, if you will just iterate them, `IEnumerable<T>` even better. But if you plan to do other list things, `IList<T>` is what we should ahve here.